### PR TITLE
Prevent array index out of range in BlockPuller. Include blocks in OnDeleteTransaction.

### PR DESCRIPTION
**Pull request by quantumagi**
Fixed some issues that popped up during work on IndexStore.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
nice, good to me

File: Stratis.Bitcoin/BlockPulling/BlockPuller.cs [hunk 1]
```diff
@@ -282,7 +282,7 @@ private void DistributeDownload(InventoryVector[] vectors, BlockPullerBehavior[]
 
             int[] scores = innernodes.Select(n => n.QualityScore == MaxQualityScore ? MaxQualityScore * 2 : n.QualityScore).ToArray();
             var totalScore = scores.Sum();
-            GetDataPayload[] getDatas = this.Nodes.Select(n => new GetDataPayload()).ToArray();
```
Comment (dangershony) [inline Stratis.Bitcoin/BlockPulling/BlockPuller.cs]: No way how did this happen? good spot

Comment (dangershony) [inline Stratis.Bitcoin/BlockPulling/BlockPuller.cs]: Are the integration tests passing?

Comment (NicolasDorier) [inline Stratis.Bitcoin/BlockPulling/BlockPuller.cs]: I was coding at 5am maybe....

**Pull request merged**